ocaml/parsing/asttypes.mli

```diff
@@ -72,3 +72,5 @@ type variance =
 type injectivity =
   | Injective
   | NoInjectivity
+
+type modality = | Modality of string [@@unboxed]
```

Comment:
I think its nice to have this type be different from `string`, especially considering there will be `type mode = | Mode of string` in the future. I considered adding an `asttypes.ml` and doing something like `jane_asttypes.ml` does, but there are a bunch of places in ppxlib where the types from `asttypes` need to be redefined as equal to the original -- so that doesn't really work either. Basically I agree that a single-constructor variant is clunky but its the only solution I could think of.

Reply:
The parser for modes/modalities may soon be more complex anyway so this point will probably be moot soon.

Reply:
I think this definition should be moved to `parsetree.mli`, as `asttypes.mli` contains only those definitions shared by parsetree and typedtree. And modalities will have a different representation in typedtree.

About the single constructor - would `private` help? (to distinguish between modes and modalities). Alternatively, since the original jane syntax doesn't distinguish the two anyway, we can keep this blurry line and fix this in a latter PR.

Reply:
BTW, I think the syntax for modalities will remains as a list of strings for a long while. But yes, we expect the syntax for mode expressions will become richer soon.

Reply:
Charlie’s point was that we can’t do a private or abstract type because asttypes and parsetree only have an mli. But I also think that I’m fine with the single-constructor variant.

Reply:
(Yes, in fact, I’ve marked this as resolved.)

Reply:
I see - but this should still be moved to `parsetree.mli`?

Reply:
makes sense to me, moved it